libcaf_openssl/src/middleman_actor.cpp

```diff
@@ -48,34 +50,53 @@ using native_socket = io::network::native_socket;
 using default_mpx = io::network::default_multiplexer;
 
 struct ssl_policy {
-  /// Reads up to `len` bytes from an OpenSSL socket.
-  static bool read_some(size_t& result, native_socket fd, void* buf,
+  ssl_policy(std::shared_ptr<ssl_session> ssl) : ssl_(ssl) {
+  }
+
+  bool read_some(size_t& result, native_socket fd, void* buf,
                         size_t len) {
     CAF_LOG_TRACE(CAF_ARG(fd) << CAF_ARG(len));
-    // TODO: implement me
-    CAF_RAISE_ERROR("unimplemented function: openssl::read_some");
+
+    return ssl_->read_some(result, fd, buf, len);
   }
 
-  static bool write_some(size_t& result, native_socket fd, const void* buf,
+  bool write_some(size_t& result, native_socket fd, const void* buf,
                          size_t len) {
     CAF_LOG_TRACE(CAF_ARG(fd) << CAF_ARG(len));
-    // TODO: implement me
-    CAF_RAISE_ERROR("unimplemented function: openssl::write_some");
+
+    return ssl_->write_some(result, fd, buf, len);
   }
 
-  static bool try_accept(native_socket& result, native_socket fd) {
+  bool try_accept(native_socket& result, native_socket fd) {
     CAF_LOG_TRACE(CAF_ARG(fd));
-    // TODO: implement me
-    CAF_RAISE_ERROR("unimplemented function: openssl::try_accept");
+
+    sockaddr_storage addr;
+    memset(&addr, 0, sizeof(addr));
+    socklen_t addrlen = sizeof(addr);
+    result = ::accept(fd, reinterpret_cast<sockaddr*>(&addr), &addrlen);
```

Comment:
Is there a conflicting declaration for `accept` that makes the `::` prefix necessary?